environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentStackConfigUpdateService.java

```diff
@@ -0,0 +1,37 @@
+package com.sequenceiq.environment.environment.service;
+
+import com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider;
+import com.sequenceiq.cloudbreak.auth.altus.Crn;
+import com.sequenceiq.cloudbreak.common.exception.NotFoundException;
+import com.sequenceiq.environment.environment.domain.Environment;
+import com.sequenceiq.environment.environment.flow.EnvironmentReactorFlowManager;
+import org.springframework.stereotype.Service;
+
+@Service
+public class EnvironmentStackConfigUpdateService {
+    private final EnvironmentService environmentService;
+
+    private final EnvironmentReactorFlowManager reactorFlowManager;
+
+    public EnvironmentStackConfigUpdateService(
+        EnvironmentService environmentService,
+        EnvironmentReactorFlowManager reactorFlowManager) {
+        this.environmentService = environmentService;
+        this.reactorFlowManager = reactorFlowManager;
+    }
+
+    public void updateAllStackConfigsByCrn(String envCrn) {
+        String accountId = Crn.safeFromString(envCrn).getAccountId();
+        if (accountId == null) {
+            throw new NotFoundException(String.format("No account id found for crn '%s'", envCrn));
+        }
```

Comment:
nit: this is not necessary as `safeFromString` will throw an exception in case crn doesn't match the pattern, which enforces the accountid can't be empty.